source/state_representation/src/trajectory/CartesianTrajectory.cpp

```diff
@@ -0,0 +1,190 @@
+#include "state_representation/trajectory/CartesianTrajectory.hpp"
+#include "state_representation/space/cartesian/CartesianState.hpp"
+
+#include "state_representation/exceptions/EmptyStateException.hpp"
+#include "state_representation/exceptions/IncompatibleReferenceFramesException.hpp"
+#include "state_representation/exceptions/IncompatibleSizeException.hpp"
+
+#include <eigen3/Eigen/src/Core/Matrix.h>
+
+namespace state_representation {
+CartesianTrajectory::CartesianTrajectory(const std::string& name, const std::string& reference_frame)
+    : TrajectoryBase<CartesianTrajectoryPoint>(name), reference_frame_(reference_frame) {
+  this->set_type(StateType::CARTESIAN_TRAJECTORY);
+  this->reset();
+}
+
+const std::string CartesianTrajectory::get_reference_frame() const {
+  return this->reference_frame_;
+}
+
+CartesianTrajectory::CartesianTrajectory(
+    const CartesianState& point, const std::chrono::nanoseconds& duration, const std::string& name,
+    const std::string& reference_frame)
+    : TrajectoryBase<CartesianTrajectoryPoint>(name), reference_frame_(reference_frame) {
+  this->set_type(StateType::CARTESIAN_TRAJECTORY);
+  this->reset();
+  CartesianTrajectoryPoint trajectory_point;
+  trajectory_point.data = point.data();
+  trajectory_point.duration = duration;
+  trajectory_point.name = point.get_name();
+  this->TrajectoryBase<CartesianTrajectoryPoint>::add_point(trajectory_point);
```

Comment:
```suggestion
  this->add_point(point, duration);
```
for the reference frame check